core/src/main/scala/org/scalarelational/model/SQLDatastore.scala

```diff
@@ -90,7 +90,7 @@ abstract class SQLDatastore protected() extends Datastore {
     b.append(column.name)
     b.append(' ')
     b.append(column.sqlType)
-    if (column.has(NotNull)) {
+    if (!column.isOptional && !column.has(Polymorphic)) {
       b.append(" NOT NULL")
```

Comment:
You need to add a test for this because this will utterly fail.  You'll end up with:
```sql
columnName = NOT NULL null
```